static/js/treatment.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+function initTreatmentDrawer()  {
+  $('#eventType').val('BG Check');
+  $('#glucoseValue').val('').attr('placeholder', 'Value in ' + browserSettings.units);
+  $('#meter').prop('checked', true);
+  $('#carbsGiven').val('');
+  $('#insulinGiven').val('');
+  $('#preBolus').val(0);
+  $('#notes').val('');
+  $('#enteredBy').val(browserStorage.get('enteredBy') || '');
+  $('#nowtime').prop('checked', true);
+  $('#eventTimeValue').val(new Date().toTimeString().slice(0,5));
+  $('#eventDateValue').val(Nightscout.utils.toDateInputValue(new Date()));
+}
+
+function treatmentSubmit(event) {
+
+  var data = {};
+  data.enteredBy = $('#enteredBy').val();
+  data.eventType = $('#eventType').val();
+  data.glucose = $('#glucoseValue').val();
+  data.glucoseType = $('#treatment-form input[name=glucoseType]:checked').val();
+  data.carbs = $('#carbsGiven').val();
+  data.insulin = $('#insulinGiven').val();
+  data.preBolus = parseInt($('#preBolus').val());
+  data.notes = $('#notes').val();
+  data.units = browserSettings.units;
+
+  var errors = [];
+  if (isNaN(data.glucose)) {
+    errors.push('Blood glucose must be a number');
+  }
+
+  if (isNaN(data.carbs)) {
+    errors.push('Carbs must be a number');
+  }
+
+  if (isNaN(data.insulin)) {
+    errors.push('Insulin must be a number');
+  }
+
+  if (errors.length > 0) {
+    window.alert(errors.join('\n'));
+  } else {
+		if ($('#othertime').is(':checked')) {
+			data.eventTime = Nightscout.utils.mergeInputTime('#eventTimeValue','#eventDateValue');
+		}
+    confirmPost(data);
+  }
+
+  if (event) {
+    event.preventDefault();
+  }
+}
+
+function confirmPost(data) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (9)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/784416116/issues/source?bid=2452379&fileBranchId=2490906#l57)